ospd/ospd.py

```diff
@@ -366,6 +371,14 @@ def process_targets_element(scanner_target) -> List:
         return OspRequest.process_target_element(scanner_target)
 
     def stop_scan(self, scan_id: str) -> None:
+        if (
+            scan_id in self.scan_collection.ids_iterator()
+            and self.get_scan_status(scan_id) == ScanStatus.QUEUED
+        ):
+            self.scan_collection.remove_file_pickled_scan_info(scan_id)
+            self.set_scan_status(scan_id, ScanStatus.STOPPED)
```

Comment:
I guess set_scan_status will call scan_collection internally? In that case maybe a new more fitting method in scan collection should be added e.g. self.scan_collection.stop_scan(scan_id) that sets the status and removed the pickled scan info.